ColinTreeListViewElement.java

```diff
@@ -1,695 +1,1422 @@
+
+    private final ArrayList<Element> elementList = new ArrayList<Element>();
+    private final ArrayList<Integer> longClickedElements = new ArrayList<Integer>();
+    private HVArrangement vaContainer = null;
+    private int currentListSize = 0;
+
+    private final Handler handler = new Handler();
+
+    private int lastClickedElement = 0;
+    private int lastLongClickedElement = 0;
+    private int lastClickedIcon = 0;
+    private int lastClickedExtraButton = 0;
+
+    // Appearance
+    private int elementHeight = 57;
+    private int elementTouchDownColor = COLOR_DEFAULT;
+    private int elementWidthBeforeIcon = 7;
+    private int elementWidthAfterIcon = 5;
+    private int elementIconWidth = 40;
+    private int elementIconHeight = 40;
+    private int elementIconShape = BUTTON_SHAPE_ROUNDED;
+    private int elementIconBgColor = COLOR_DEFAULT;
+    private int elementIconTextColor = COLOR_DEFAULT;
+    private float elementIconTextFontSize = 24;
+    private boolean elementIconTextFontBold = true;
+    private boolean elementIconMultiParams = false;
+    private int elementIconPaddings = 0;
+    private int elementTextColor = COLOR_BLACK;
+    private float elementTextFontSize = 14f;
+    private boolean elementTextFontBold = false;
+    private int elementTextHeight = 18;
+    private int elementSubTextColor = COLOR_LTGRAY;
+    private float elementSubTextFontSize = 12;
+    private boolean elementSubTextFontBold = false;
+    private int elementSubTextHeight = 18;
+    private int elementWidthBeforeExtraButton = 2;
+    private String elementExtraButtonText = "";
+    private float elementExtraButtonTextFontSize = 12;
+    private boolean elementExtraButtonTextFontBold = false;
+    private int elementExtraButtonWidth = 20;
+    private int elementExtraButtonHeight = 20;
+    private int elementExtraButtonPaddings = 0;
+    private int elementExtraButtonBgColor = COLOR_LTGRAY;
+    private int elementExtraButtonShape = BUTTON_SHAPE_OVAL;
+    private String elementExtraButtonImage = "";
+    private int elementUnderlineColor = COLOR_LTGRAY;
+    private int elementUnderlineWidth = 1;
+
+    // Behaviour
+    private boolean scrollBottomAfterAdd = false;
+    private boolean asyncImageLoad = false;
+    private boolean cacheImage = false;
+    private boolean extraButtonEnabled = false;
+    private boolean multiSelection = false;
+    private boolean showTouchDownColor = true;
+
+    private final HashMap<String, CachedImage> iconMap = new HashMap<String, CachedImage>();
+
+    private final Form form;
+
+
+    private static YailList makeYailList(Object... obj) {
+        return YailList.makeList(obj);
+    }
+
+
+    public ColinTreeListView(ComponentContainer container) {
         super(container.$form());
+        form = container.$form();
+        Log.d(LOG_TAG, "ColinTreeListView Created");
+    }
+
+    @SimpleFunction
+    public void Initialize(VerticalArrangement verticalArrangement) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Reply:
Hi,
Naming conventions of Extensions are a little bit different than those in java.
You would like to check this:
https://community.appinventor.mit.edu/t/naming-conventions-for-extension-developers/22186

Reply:
noticed